### PR TITLE
Add follow and unfollow endpoints

### PR DESCRIPTION
**Pull request by sercanersoy**
### Description
Follow and unfollow endpoints are added to backend

### Related Issues
See #171 

### Other Comments
Try like the following:

POST --> localhost:8000/users/1/followings
  data: {"following_pk": 2}

GET --> localhost:8000/users/1/followings

GET --> localhost:8000/users/2/followers

DELETE --> localhost:8000/users/1/followings/2

Don't forget to migrate your db.

### Checklist
- [x] Code runs
- [ ] Created tests (if applicable)
- [ ] All tests passing
- [ ] Extended the README and the documentation (if applicable)

### CODE REVIEW

File: nova/nova/serializers.py [hunk 1]
```diff
@@ -28,7 +29,3 @@ def create(self, validated_data):
         validated_data['password'] = make_password(validated_data.get('password'))
 
         return super(UserSerializer, self).create(validated_data)
-
-    def update(self, instance, validated_data):
```
Comment (alpercakan) [inline nova/nova/serializers.py]: Why is this removed

File: nova/nova/views.py [hunk 2]
```diff
@@ -71,4 +126,4 @@ def auth_tokens_coll(request):
         raise AuthenticationFailed()
 
     token, created = Token.objects.get_or_create(user=user)
-    return Response({'token': token.key, 'user_id': user.pk})
+    return Response({'token': token.key, 'user_id': user.pk}, status=status.HTTP_200_OK)
```
Comment (alpercakan) [inline nova/nova/views.py]: I think the default status is 200, you don't need to pass it. If you added this part for the sake of explicitness ,then it's alright

File: nova/nova/views.py [hunk 3]
```diff
@@ -61,6 +59,63 @@ def user_res(request, pk):
...
+@api_view(['DELETE'])
+def user_following_res(request, user_pk, following_pk):
+    try:
+        user = User.objects.get(pk=user_pk)
+        following = User.objects.get(pk=following_pk)
+    except User.DoesNotExist:
+        raise NotFound()
+
+    if request.method == 'DELETE':
+        if request.user.pk != user_pk:
```
Comment (alpercakan) [inline nova/nova/views.py]: This is great! I'd forgotten to add a similar check to the update user endpoint (you should be able to update a user iff you are that user or you are an admin.) Maybe we can create an issue about that.

File: nova/nova/serializers.py (hunk 1 above)
Comment (sercanersoy) [reply nova/nova/serializers.py]: Because when I update a user, it was encrypting the encrypted password again. But if the user changes his/her password, it should be encrypted before it is stored, so you are right in that point. I will correct it, thanks!

File: nova/nova/views.py (hunk 3 above)
Comment (sercanersoy) [reply nova/nova/views.py]: I added this check to the PUT request in your migration PR. If more change is needed like this, I agree that we can open an issue for that.

Comment (alpercakan) [reply nova/nova/views.py]: Oh hadn't seen that. Thanks!
